ui/src/app/base/selectors/general/machineActions.js

```diff
@@ -6,4 +8,22 @@ import { generateGeneralSelector } from "./utils";
 
 const machineActions = generateGeneralSelector("machineActions");
 
+/**
+ * An intermediary selector to get and pass params to the selector.
+ * @param {Object} state - The redux state.
+ * @returns {Array} The provided params.
+ */
+const getParams = (state, ...params) => [...params];
+
+/**
+ * Get a machine action by name.
+ * @param {Object} state - The redux state.
+ * @param {String} name - The name of a machine action.
+ * @returns {Object} A machine action.
+ */
+machineActions.getByName = createSelector(
+  [machineActions.get, getParams],
+  (items, [name]) => items.find((item) => item.name === name)
```

Comment:
`items` is too generic, I think `actions` or similar would work better. Also I'm finding this `getParams` pattern confusing and hard to read (though I guess the `createSelector` api is a little weird to begin with). I'm wondering if it's too early to abstract it out, or if we want to abstract it out at all. Is it possible for it to be something like:
``` js
machineActions.getByName = createSelector(
  [machineActions.get, (state, name) => name],
  (actions, name) => actions.find((action) => action.name === name)
);
```

Actually does it even need to be its own selector to begin with? Couldn't you just use `machineActions.get ()`then `.find()` the action in the component? I'm not actually sure if there are any benefits to using selectors if they have dynamic arguments.

Reply:
The benefit of doing it this way is we can use `createSelector` and it'll memoize. Unfortunately we have to do it this way because we don't use props to get the data into the components so the selectors can't use those for params.